### ISSUE TITLE
Building UE5.3 container fails due to 'Bridge' Plugin failed to load

### ISSUE THREAD

**Issue by petergerten** [adamrehn/ue4-docker#337]
```
ue4-docker version:         0.0.110 (latest available version is 0.0.110)
Operating system:           Linux (Arch Linux, 6.1.50-1-lts)
Docker daemon version:      24.0.5
NVIDIA Docker supported:    Yes
Maximum image size:         No limit detected
Available disk space:       842.41 GiB
Total system memory:        125.64 GiB physical, 0 bytes virtual
CPU:                        24 physical, 48 logical ()
```

Build Settings:
```
[ue4-docker build] COMMAND-LINE INVOCATION:
[ue4-docker build] ['ue4-docker', '--cuda=12.2.2', '-basetag', 'ubuntu20.04', '5.3.1']
[ue4-docker build] UNREAL ENGINE VERSION SETTINGS:
[ue4-docker build] Custom build:  No
[ue4-docker build] Release:       5.3.1
[ue4-docker build] Repository:    https://github.com/EpicGames/UnrealEngine.git
[ue4-docker build] Branch/tag:    5.3.1-release
[ue4-docker build] ADVANCED CONFIGURATION OPTIONS:
[ue4-docker build] credential_mode: "secrets"
[ue4-docker build] excluded_components: {"ddc": false, "debug": false, "templates": false}
[ue4-docker build] gitdependencies_args: "--exclude=Android --exclude=Mac --exclude=Win32 --exclude=Win64"
[ue4-docker build] LINUX CONTAINER SETTINGS
[ue4-docker build] Base OS image: nvidia/cuda:12.2.2-devel-ubuntu20.04
```

Error:
```
6471.4 LogCore: Warning: dlopen failed: libgbm.so.1: cannot open shared object file: No such file or directory
6471.4 LogPluginManager: Error: Plugin 'Bridge' failed to load because module 'Bridge' could not be loaded.  There may be an operating system error or the module may not be properly set up.
6471.4 Message dialog closed, result: Ok, title: Message, text: Plugin 'Bridge' failed to load because module 'Bridge' could not be loaded.  There may be an operating system error or the module may not be properly set up.
6471.4 LogCore: Engine exit requested (reason: EngineExit() was called)
```

I tried the following combinations all resulting in the same error:
- UE 5.3.1 + Ubuntu 22.04
- UE 5.3.1 + Ubuntu 20.04
- UE 5.3.0 + Ubuntu 20.04

I found that the Dockerfile hints to potential problems which Quixel Bridge here:
https://github.com/adamrehn/ue4-docker/blob/edb8be4f25f866b1f740cfcdd71b00409e8770c5/src/ue4docker/dockerfiles/ue4-build-prerequisites/linux/Dockerfile#L44
But the updated Dockerfile is already being used...

Searching for the error lead me to this: https://www.reddit.com/r/unrealengine/comments/16e7z39/plugin_bridge_failed_to_load_because_module/
Here somebody solved this by updating the Quixel Plugin. Not sure if this would also be relevant here?

**Comment by slonopotamus** [comment]
Just remove Bridge plugin from your .uproject. Installed engine builds do not contain Bridge plugin, so UE cannot load it when opening your project.

**Comment by petergerten** [comment]
@slonopotamus I am actually not building a project but only want to build the container with 
`ue4-docker build --cuda=12.2.2 -basetag ubuntu22.04 5.3.1`

This is when it happens. Or am I missing something here?

**Comment by slonopotamus** [comment]
Err... Right. Looks like my data is a bit outdated) Possibly the issue is missing `libgbm.so.1`.

**Comment by adamrehn** [comment]
Diagnosing this issue was quite interesting!

Back when I first [added the packages needed by CEF](https://github.com/adamrehn/ue4-docker/commit/3f5b10e987418b07212b53ba3b1b4c991f28c6ad) (and in turn by plugins that depend on CEF, such as Bridge), ue4-docker used the [nvidia/cudagl](https://hub.docker.com/r/nvidia/cudagl) base image to provide CUDA support, which extends the [nvidia/opengl](https://hub.docker.com/r/nvidia/opengl) base image and therefore provides all of the same packages. My methodology for identifying CEF dependency packages was simple: use `ldd` to determine which `.so` files were missing, and add the packages that provide those files. Since the same common files were present in both of the upstream base images, the set of packages that I added to the Dockerfile worked nicely when testing both OpenGL and CUDA builds.

Much more recently in #315, ue4-docker switched to using the [nvidia/cuda](https://hub.docker.com/r/nvidia/cuda) base image for CUDA support, since the old `nvidia/cudagl` image no longer appears to be maintained. (This has [caused headaches](https://github.com/adamrehn/ue4-runtime/blob/137915b1185ea74c3dc14688e5efa6d38cace7af/build.py#L107-L109) for the ue4-runtime project as well.) As a result, CUDA builds no longer include the packages from the `nvidia/opengl` base image. Specifically, CUDA builds no longer [include libegl1-mesa-dev](https://gitlab.com/nvidia/container-images/opengl/-/blob/ubuntu22.04/glvnd/devel/Dockerfile?ref_type=heads#L9), which has the following dependency chain as reported by the [apt-rdepends](https://manpages.ubuntu.com/manpages/jammy/en/man1/apt-rdepends.1.html) tool:

```
libegl1-mesa-dev -> libegl-dev -> libegl1 -> libegl-mesa0 -> libgbm1
```

As a result, the `libgbm1` package is not present, and so the file `libgbm.so.1` is missing. I'll add an explicit entry for `libgbm1` in the package list to ensure it is present in both OpenGL and CUDA builds.

**Comment by petergerten** [comment]
Yes, indeed. I manually edited the generated Dockerfile and added
`apt-get install -y libgbm-dev`

Now the build succeeds.

**Comment by adamrehn** [comment]
Excellent! In that case, I'll close this issue as fixed by commit d126578.
